### ISSUE TITLE
Downloading a driveItem file does not work for driveItems extracted from Sharepoint site

### ISSUE THREAD

**Issue by 3vap** [vgrem/Office365-REST-Python-Client#798]
Initially I tried downloading a file which is a driveItem extracting it from url:

file_item = client.shares.by_url(file_abs_url).drive_item.get().execute_query()

then using the example for downloading large files - this works.

However when I extract the driveITem using a different technic:

list_of_items=drive_item_folder.children.get().execute_query()

I get list of driveItem files which produce error when applying the same downlaod function:

office365.runtime.client_request_exception.ClientRequestException: ('BadRequest', "Resource not found for the segment '...'.", '400 Client Error: Bad Request for url: https://graph.microsoft.com/v1.0/shares/...

Then I tested what happens when I compare those drive Items for the file: 1st driveItem extracted from sharable link, 2nd from the children function and they are not the same.

They got same fields values for @microsoft.graph.downloadUrl, createdDateTime, eTag, id, name, lastModifiedDateTime, webUrl, 

But different objects in parentReference, file, fileSystemInfo, shared.

To sum up: not all driveItems are possible to download, even when they are derived from the same file. Sharable link for the file works, but extracting driveItem from children list of parent folder driveItem fails to work even though they are items for the same file ...

**Comment by 3vap** [comment]
I've entered the @microsoft.graph.downloadUrl into web browser and the file got downloaded.

So it seems the problems is with the function:

file_item.download_session( local_file, print_progress, chunk_size=1024 * 512 ).execute_query()

It operates on thte same driveItem having @microsoft.graph.downloadUrl  that is possible to download via url but not using the method.

**Comment by 3vap** [comment]
Using the single file approach:

remote_file.download(local_file).execute_query()

 gives the same 400 error:

office365.runtime.client_request_exception.ClientRequestException: ('BadRequest', "Resource not found for the segment '...'.", '400 Client Error: Bad Request for url: https://graph.microsoft.com/v1.0/shares/...

**Comment by vgrem** [comment]
Same root cause as #993 - confirmed still present. `DriveItem.download` and `download_session` build the content URL from the item's `resource_path` rather than from `parentReference.driveId` + `id`. When an item is fetched via `children.get()` in a SharePoint-connected drive, the `resource_path` resolves incorrectly through the `shares` endpoint, generating:

```
/shares/{encoded_url}/driveItem:/filename:/content  ← broken
```

Instead of:
```
/drives/{driveId}/items/{itemId}/content            ← correct
```

**Workaround - use `@microsoft.graph.downloadUrl` directly:**

```python
items = folder.children.get().execute_query()
for item in items:
    if item.is_file:
        download_url = item.properties.get("@microsoft.graph.downloadUrl")
        if download_url:
            import requests
            response = requests.get(download_url)
            with open(item.name, "wb") as f:
                f.write(response.content)
```

`@microsoft.graph.downloadUrl` is a pre-authenticated short-lived URL that works directly without the library's URL building - bypasses the broken path resolution entirely.

**Or resolve to native path first:**

```python
item = drive_item_folder.children.get().execute_query()[0]
native = client.drives[item.parent_reference.drive_id]\
    .items[item.id].get().execute_query()

with open("file.pdf", "wb") as f:
    native.download(f).execute_query()
```

Fix tracked in v3.0.0 alongside #993 - `download` will use `ensure_properties(["id", "parentReference"])` to resolve canonical path before building content URL. Related: #993.
